File: task-03/Elixir/4/Sub-Task4.exs
defmodule DiamondPattern do
  def generate_diamond(n) do
    upper_part = for i <- 0..(n-1) do
      spaces = String.duplicate(" ", n - i - 1)
      stars = String.duplicate("*", 2 * i + 1)
      spaces <> stars
    end

    lower_part = for i <- (n-2)..0 do
      spaces = String.duplicate(" ", n - i - 1)
      stars = String.duplicate("*", 2 * i + 1)
      spaces <> stars
    end

    Enum.join(upper_part ++ lower_part, "\n")
  end

  def main do
    case File.read("input.txt") do
      {:ok, content} ->
        n = String.trim(content) |> String.to_integer()

        diamond_pattern = generate_diamond(n)

        File.write!("output.txt", diamond_pattern)

        IO.puts("Pattern printed to output.txt")

      {:error, reason} ->
        IO.puts("Error reading input file: #{reason}")
    end
  end
end

DiamondPattern.main()

File: task-03/Python/4/Sub-Task4.py
def generate_diamond(n):
    lines = []
    for i in range(n):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        lines.append(line)
    
    for i in range(n - 2, -1, -1):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        lines.append(line)
    
    return '\n'.join(lines)

def main():
    with open('input.txt', 'r') as file:
        n = int(file.read().strip())
    
    diamond_pattern = generate_diamond(n)
    
    with open('output.txt', 'w') as file:
        file.write(diamond_pattern)

if __name__ == '__main__':
    main()
